File: RestMongoAggregations/data/flight.go
package data

import (
	"encoding/json"
	"io"
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Flight struct {
	ID         primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Date       time.Time          `bson:"date" json:"date"`
	EndPlace   string             `bson:"endPlace" json:"endPlace"`
	StartPlace string             `bson:"startPlace" json:"startPlace"`
	Capacity   int                `bson:"capacity" json:"capacity"`
	Price      int                `bson:"price" json:"price"`
	FreeSeats  int                `bson:"freeSeats" json:"freeSeats"`
}

type Flights []*Flight

func (f *Flights) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(f)
}

func (f *Flight) FromJSON(r io.Reader) error {
	d := json.NewDecoder(r)
	return d.Decode(f)
}

func (f *Flight) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(f)
}

File: xmlFrontend/src/app/modules/model/flight.ts
export class Flight{
    flightId: String = "";
    date!: string;
    endPlace: String = "";
    startPlace: String = "";
    capacity: number = 0;
    price: number = 0;
    freeSeats: number = 0;
    
    public constructor(obj?: any) {
        if (obj) {
            this.flightId = obj.FlightId;
            this.date = obj.Date;
            this.endPlace = obj.EndPlace;
            this.startPlace = obj.StartPlace;
            this.capacity = obj.Capacity;
            this.price = obj.Price;
            this.freeSeats = obj.FreeSeats;
        }
    }
}